### PR TITLE
Add docker file support for deploy 

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes https://github.com/dotnet/tye/issues/146

Adds a new type that represents a DockerFile (so it gets ports and stuff).

### CODE REVIEW

**Comment by rynowak** [comment]
Is the idea that we'll generate a k8s deployment/service for a dockerfile pretty much the same we we would for a project? SPICY.

**Comment by jkotalik** [comment]
Yep, only for files that have a docker file though, not just random images.

**Comment by davidfowl** [comment]
> Is the idea that we'll generate a k8s deployment/service for a dockerfile pretty much the same we we would for a project? SPICY.

I'm worried about the versioning/tagging.

File: src/Microsoft.Tye.Core/DockerFileServiceBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.Tye
+{
+    public class DockerFileServiceBuilder : ProjectServiceBuilder
```
Comment (rynowak) [inline src/Microsoft.Tye.Core/DockerFileServiceBuilder.cs:7]: What does this represent? Is this a .NET project with a Dockerfile?

Or is this meant to capture a non-project that has a Dockerfile?

Bascially the use of inheritance seems like a convenient hack.

Comment (jkotalik) [reply src/Microsoft.Tye.Core/DockerFileServiceBuilder.cs:7]: > Or is this meant to capture a non-project that has a Dockerfile?

This one.

It was a very convenient "refactoring".

**Pull request merged**